src/index.js

```diff
@@ -12,12 +12,17 @@ const parseLocale = (preset, object, isLocal) => {
   let l
   if (!preset) return L
   if (typeof preset === 'string') {
-    if (Ls[preset]) {
-      l = preset
+    const presetLower = preset.toLowerCase()
+    if (Ls[presetLower]) {
+      l = presetLower
     }
     if (object) {
-      Ls[preset] = object
-      l = preset
+      Ls[presetLower] = object
+      l = presetLower
+    }
+    const presetSplit = preset.split('-')
+    if (!l && presetSplit.length > 1) {
```

Comment:
if (!l && presetSplit.length) {

we could just check the length is above 0

Reply:
That was my first attempt but other unit tests break if I don't check that the split actually splits the preset string (i.e it results in more than one array element).

The test "Not supported locale string fallback to previous one (instance)" in locale.test.js fails with "if (!l && presetSplit.length) {"

The "not_supported_locale_string" is split into one element => recursive call, "not_supported_locale_string" is split into one element again => recursive call ... etc

Reply:
oh. I see, you can just change the "not_supported_locale_string" test string 😁

Reply:
You mean like "not-supported-locale-string"?
But that string will never split into something valid.
I still think it's a good idea to only recurse if the preset actually splits into 2 parts.